docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
+The view command will search for an event by its name and display all its participants if found.
+It is implemented in the `ViewCommand` class which extends the base `Command` class and interacts with the `EventList` to retrieve and display participant information.
+
+The `ViewCommand` class handles one primary function:
+
+1. It is to enable users to view all participants associated with a specific event. 
+It accomplishes this by searching for the event by its name and displaying participant details if the event is found.
+
+#### Feature Implementation
+
+Given below is an example usage scenario for the `view` mechanism, and how it behaves at each step
+
+1. The user enters the command `view` followed by an `-e` flag to indicate which event to view participants from
+2. The command is processed by the `Parser`, which identifies that the user intends to view participants for the specified event.
+3. The `ViewCommand` attempts to retrieve the event from the `EventList` using the `getEventByName` method.
+4. If the event is found:
+   - The `ViewCommand` constructs a success message that includes the number of participants and their details in a formatted manner.
+
+   If the event is not found:
+   - The `ViewCommand` sets an error message.
+5. The message generated by `ViewCommand` is returned to the user interface, which then displays it to the user
+
+The interactions between components during the execution of the `view` command are show in the **Sequence Diagram** below:
+
+<img src = "images/ViewEventSequenceDiagram.png">
+
+### Mark/unmark feature
+
+The `mark/unmark` feature allows users to mark events as done or not done. The feature comprises `MarkEventCommand`, which 
+extends `Command`. This class performs one operation, which marks a specified event as done or not done.
+
+The above operation is implemented as `MarkEventCommand#execute()`. This overrides the `Command#execute()` operation in `Command`,
+and is invoked when the latter operation is called.
+
+#### Feature implementation
+
+Given below is an example usage scenario for the mark/unmark mechanism, and how it behaves at each step.
+
+1. The user adds an event `Event 1` to the event list. The mark status for `Event 1` is initially `false` or not done.
+
+2. The user enters the command `mark -e Event 1 -s done` to mark `Event 1` as done. `MarkEventCommand` calls `MarkEventCommand#execute`,
+in which it gets the event `Event 1` from the event list, and sets its mark status to `true` or done.
+
+3. The user then enters the command `mark -e Event 1 -s undone` to mark `Event 1` as not done. The `MarkEventCommand` again calls `MarkEventCommand#execute`,
+in which it gets the event `Event 1` from the event list, and sets its mark status to `false` or not done.
+
+The interactions between components during the execution of the `mark` command are show in the **Sequence Diagram** below:
+
+<img src = "images/MarkEventSequenceDiagram.png">
```

Comment:
I think the boolean variable toMark can be renamed to sound like booleans (such as isDone) so that there is no coding standard violations